src/acp/util/setPublicAccess.test.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright 2021 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it, expect } from "@jest/globals";
+import { ACL, ACP } from "../constants";
+import {
+  DEFAULT_ACCESS_CONTROL_RESOURCE_URL,
+  TEST_URL,
+} from "../mock/constants";
+import { createDatasetFromSubjects } from "../mock/dataset";
+import { mockAccessControlledResource } from "../mock/mockAccessControlledResource";
+import { setPublicAccess } from "./setPublicAccess";
+
+describe("setPublicAccess()", () => {
+  it("sets an access mode for the Public", async () => {
```

Comment:
Do we have a tests verifying that `setXAccess` overrides existing access ?

Reply:
The setXAccess functions are adding and removing access only in the default elements.

That is tested.

I think it is an alright (and documented) limitation, that is at least until we properly support external resources consistently in access control. And even if we did that, I am not sure we'd want to automatically update all the conflicting access rather than produce an output hinting at the conflicts to be addressed.

Reply:
Ah right sorry, I forgot that we were working under the assumption that these access only apply to the default elements. That's alright then for now, as indeed a different approach requires more discussion.